docs/guide/account_and_client.rst

```diff
@@ -14,6 +14,18 @@ Here is an example:
     :dedent: 4
 
 
+Custom nonce logic
+------------------
+
+By default, :ref:`Account` calls StarkNet for nonce every time a new transaction is signed or executed.
+This is okay for most users, but in case your applications needs to pre-sign multiple transactions
+for execution, deals with high amount of transactions or just needs to support different nonce
+logic, it is possible to so with :ref:`Account`. Simply overwrite the
+:meth:`~starknet_py.net.account.account.Account.get_nonce` method with your own logic.
+
+.. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_custom_nonce.py
+    :language: python
+    :dedent: 4
```

Comment:
Maybe we should put it at the end of this section? I feel like it's not that common to use use it, and it is the second subsection in the whole guide.
Or we could put it somewhere in `signing` section maybe?

Reply:
We can move it to the end.